paramiko/pkcs11.py

```diff
@@ -0,0 +1,115 @@
+from ctypes import (c_void_p, c_ulong, c_int, c_char_p, cast, addressof,
+                    sizeof, byref, cdll, Structure)
+import subprocess
+import os
+import errno
+from paramiko.ssh_exception import SSHException
+
+
+def pkcs11_get_public_key():
+    public_key = ""
+    try:
+        p = subprocess.Popen(["pkcs15-tool", "--read-ssh-key", "01"],
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE,
+                             stdin=subprocess.PIPE)
+        out, err = p.communicate()
+        if out is not None:
+            public_key = out
+    except OSError as error:
+        if error.errno == errno.ENOENT:
+            raise SSHException("Cannot find pkcs15-tool in PATH.")
```

Comment:
If not translating to this exception, you still need to re-raise the original exception, e.g. just

```python
else:
    raise
```